docs/developers/benchmark-testing.md

```diff
@@ -20,5 +20,38 @@ If you want to cause failure in the case of regression, also add "–-benchmark-
 
 To compare two already stored results, the standalone command "pytest-benchmark compare" followed by the paths of the files you wish to compare can do that. Unlike the previous method, these files do not need to be stored in any specific folder.
 
+For further options, see https://pytest-benchmark.readthedocs.io/en/latest/usage.html
 
-For further options, see https://pytest-benchmark.readthedocs.io/en/latest/usage.html
\ No newline at end of file
+
+## CI Integration
+
+The CI workflow is designed to perform benchmark tests on direct pushes to branches. It uses `pytest` with benchmark plugins to assess performance regressions or improvements. To accommodate varying performance expectations across features or branches, a JSON file named `benchmark_thresholds.json` is used to specify custom benchmark thresholds.
+
+## How Benchmark Testing Works
+
+Benchmark tests are run using `pytest` configured to only execute benchmark-related tests:
+
+- **Flag `--benchmark-only`**: Restricts `pytest` to run only benchmark tests.
+- **Flag `--benchmark-json`**: Outputs the results to a JSON file, allowing for subsequent comparisons.
+- **Flag `--benchmark-compare`**: Compares current benchmarks against the last saved benchmarks.
+- **Flag `--benchmark-compare-fail`**: Specifies a threshold for test failure; if performance degrades beyond this percentage, the test fails.
+
+These benchmarks help ensure that new code does not introduce significant performance regressions.
+
+## Setting Benchmark Thresholds
+
+The benchmark threshold can be defined for each branch to accommodate specific performance criteria associated with different types of work (e.g., feature development, bug fixes). This threshold is set in the `benchmark_thresholds.json` file. If not set, threshold is set to 10 (10%) by default.
+
+### Example `benchmark_thresholds.json`:
+
+```json
+{
+  "feature/cool-feature": 20,
+  "bugfix/urgent-fix": 15
+}
+```
+
+## Artifact Handling
+
+- **Uploading Artifacts**: After benchmarks are run, the resulting `0001_output.json` file is uploaded as an artifact to GitHub Actions, allowing it to be used in future benchmark comparisons.
```

Comment:
suggestion: adding link to the already uploaded artifacts (from some older GitHub Actions build as an example) would be very useful here